bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/DecimalType.java

```diff
@@ -165,6 +165,9 @@ public State as(Class<? extends State> target) {
             } else {
                 return UnDefType.UNDEF;
             }
+        } else if (target == HSBType.class) {
+            return new HSBType(DecimalType.ZERO, PercentType.ZERO,
+                    new PercentType(this.toBigDecimal().multiply(new BigDecimal(100))));

```

Comment:
[This code](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/math/BigDecimal.java#l1207) shows that any ints between 0 and 10, inclusive, are better to use the `valueOf(long)` static factory method to avoid unnecessary construction of new BigDecimals.  But remaining ignorant of the current implementation of `valueOf` and just going by the JavaDoc's advice seems sensible, should later/other implementations ever cache other values outside that range based on frequent usage (however unlikely we think that might be).  I think there is no `valueOf(int)` static factory method because the `int` is just widened to a `long` without side effects.